packages/legend-application-query/src/__lib__/LegendQueryUserDataHelper.ts

```diff
@@ -45,12 +47,43 @@ export class LegendQueryUserDataHelper {
               {
                 data,
               },
-            ) as { data: RecentlyViewedQueriesData }
+            ) as { data: SavedData }
           ).data,
       ) ?? []
     );
   }
 
+  static persistValue(
+    service: UserDataService,
+    value: string,
+    persistedData: SavedData,
+    opts: {
+      key: LEGEND_QUERY_USER_DATA_KEY;
+      limit: number;
+    },
+  ): void {
+    const idx = persistedData.findIndex((data) => data === value);
+    if (idx === -1) {
+      if (persistedData.length < opts.limit) {
+        persistedData.unshift(value);
+      } else {
+        persistedData.pop();
+        persistedData.unshift(value);
+      }
```

Comment:
Nit: could simplify this to
```suggestion
      if (persistedData.length >= opts.limit) {
        persistedData.pop();
      }
      persistedData.unshift(value);
```